src/Eventso.Subscription.Hosting/PoisonEventQueueRetryingService.cs

```diff
@@ -0,0 +1,103 @@
+using System.Collections.Frozen;
+using System.Runtime.CompilerServices;
+using Confluent.Kafka;
+using Eventso.Subscription.Kafka;
+using Eventso.Subscription.Kafka.DeadLetter;
+
+namespace Eventso.Subscription.Hosting;
+
+public sealed class PoisonEventQueueRetryingService : IPoisonEventQueueRetryingService
+{
+    private readonly IReadOnlyCollection<Worker> _workers;
+
+    public PoisonEventQueueRetryingService(
+        IEnumerable<ISubscriptionCollection> subscriptions,
+        IMessagePipelineFactory pipelineFactory,
+        IMessageHandlersRegistry handlersRegistry,
+        IPoisonEventQueueFactory poisonEventQueueFactory,
+        ILoggerFactory loggerFactory)
+    {
+        _workers = subscriptions
+            .SelectMany(x => x)
+            .SelectMany(x => x.ClonePerConsumerInstance())
+            .Select(x => CreateTopicRetryingService(
+                x,
+                pipelineFactory,
+                handlersRegistry,
+                poisonEventQueueFactory,
+                loggerFactory))
+            .ToArray();
+    }
+
+    public async Task Run(CancellationToken token)
+    {
+        if (_workers.Count == 0)
+            return;
+
+        var workerTasks = _workers.Select(r => r.Run(token)).ToArray();
+        await foreach (var _ in AsyncEnumerableEx.Merge(workerTasks).WithCancellation(token))
+        {
+            // do nothing
+        }
```

Comment:
Don't see a reason why this should be anything more complicated than:
```suggestion
        await Task.WhenAll(_workers.Select(r => r.Run(token)));
```

without any need for overall discarded consume results

Reply:
probably i had some plans for every result but obv it is not needed anymore